src/main/scala/viper/silver/verifier/VerificationError.scala

```diff
@@ -154,17 +154,15 @@ trait ErrorMessage {
 trait VerificationError extends AbstractError with ErrorMessage {
   def reason: ErrorReason
   def readableMessage(withId: Boolean = false, withPosition: Boolean = false): String
-  override def readableMessage : String = {
-    val rm : String = readableMessage(false, true)
-    if (counterexample.isDefined){
-      rm + "\n" + counterexample.get.toString
-    }else{
-      rm
-    }
-  }
+  override def readableMessage : String = (readableMessage(false, true) + failureContexts.map(e => e.toString).mkString("\n"))
```

Comment:
* Not your "fault", but please change to `readableMessage: String` (whitespace)
* Drop the parentheses surrounding the right-hand side expression